### PR TITLE
Feature/map 47/endpoint for personal data

### PR DESCRIPTION
**Pull request by 6mahdapihka9**
also created a profile page for testing.

to test: follow this link http://localhost:3000/test-profile-data

insert id in input field and short info about the user will appear in a container below

current available ids:
627133cdfdc359aca5dc260a
627133d0fdc359aca5dc260d
627133ecfdc359aca5dc2610
62713475ef5735f85c4d3076
62713e486322d7c81757c177

### CODE REVIEW

File: backend/controllers/UserController.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Response, Request } from "express";
+import User from "../models/UserModel";
+
+const UserController = {
+  async getUserData(req: Request, res: Response) {
+    try {
+      const userToken = req.params.userToken;
```
Comment (steadyGuy) [inline backend/controllers/UserController.ts]: We will get token from the header (Bearer Token)
const token = req.header('Authorization')?.split(' ')[1];

File: backend/controllers/UserController.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+      const userData = (await User.find(
+          {
+            userToken: userToken
```
Comment (steadyGuy) [inline backend/controllers/UserController.ts]: An email is stored in a token. Just get an email from token using jsonwebtoken package and secret key (stored in process. env) We shouldn't store tokens inside our DB, as it expire in one day.

File: backend/models/UserModel.ts [hunk 3]
```diff
@@ -8,6 +8,18 @@ export interface IUser extends Document {
   createdAt: Date | string;
   updatedAt: Date | string;
   isValidPassword: (pass: string) => boolean;
+
+  displayName: string;
+  description: string;
+  imageUrl: string;
+  userToken: string;
```
Comment (steadyGuy) [inline backend/models/UserModel.ts]: I think, we souldn't store token.

File: backend/models/UserModel.ts [hunk 4]
```diff
@@ -21,6 +33,44 @@ const schema = new mongoose.Schema(
       type: String,
       required: true,
     },
+
+    userToken: {
+      type: String,
+      required: true,
+    },
```
Comment (steadyGuy) [inline backend/models/UserModel.ts]: same

File: backend/routes/index.ts [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import express from "express";
 import FilesController from "../controllers/FilesController";
+import UserController from "../controllers/UserController";
 import LocationsController from "../controllers/LocationsController";
 import AuthController from "../controllers/AuthController";
 import passport from "../libs/passport";
 
 const router = express.Router();
 
+router.get("/userData/:userToken", UserController.getUserData);
```
Comment (steadyGuy) [inline backend/routes/index.ts]: req.params.userToken. instead, better use _id, and find by it. I don't understand the purpose of token here, as it's revealed in browser query (which is not secure).

**Review by steadyGuy — CHANGES_REQUESTED** [review]

File: backend/controllers/UserController.ts (hunk 1 above)
Comment (6mahdapihka9) [reply backend/controllers/UserController.ts]: changed to fetch and find by id now

Comment (6mahdapihka9) [reply backend/controllers/UserController.ts]: and pass by param

how can i get id of user from your access token?

**Review by steadyGuy — APPROVED** [review]

**Review by Yojuks — APPROVED** [review]

**Pull request merged**